### DISCUSSION TITLE
Splash screen during start up?

### DISCUSSION THREAD

**Discussion by jjw24** [Flow-Launcher/Flow.Launcher#660] [Ideas]
Should we add a splash screen while flow is starting up? 

Begin splash at the very beginning of starting up and will fade once startup loading is finished. 

Maybe another option for toggling 'show splash screen on startup' for those who like to keep it simple.

The splash screen can simply show our logo.

**Comment by pc223** [comment]
Maybe we should try to do no delay startup, show query box immediately after launch while waiting for loading, maybe show loading progress as a result item but still let User typing 💭

**Reply by taooceros** [reply]
Related to https://github.com/Flow-Launcher/Flow.Launcher/issues/391

**Reply by jjw24** [reply]
Letting user use it when it's not ready I don't think it's a good user experience especially for new users, it becomes unpredictable when accuracy of the result is relied on, e.g. searching for a folder or file to see if it exists. Most apps would display splash screen then fade and the program is ready to go, otherwise it feels hacky/clunky.

**Reply by jjw24** [reply]
Is there an example app maybe web page that has implemented something similar like this I can try out?

**Reply by pc223** [reply]
Splash screen is pretty annoying, I prefer see the real UI immediately, init stuff can do in the background, on my machine the startup time is only 2 secs, when users done typing 3-4 characters the init already done. We can do something like only send query when all plugins done init so no fear about accuracy.

Time consuming first time init like indexing of Programs.Plugin (takes 0.5->2min), ideally, should be done in installing proces or right after installing.
